client/src/hooks/customHooks/useHistory.ts

```diff
@@ -0,0 +1,9 @@
+import { useContext } from 'react';
+import { HistoryContext } from '../context';
+
+const useHistory = () => {
+  const { history } = useContext(HistoryContext);
+  return history;
```

Comment:
```suggestion
  const { setLocation } = useContext(HistoryContext);
  
  const push = (path: string) => {
    history.pushState('', '', path);
    setLocation(path);
  }
  
  const back = () => {
    history.back()
  }
    
  return { push, back };
```
이전에 말했듯이 이렇게 구현해볼 수 있겠군요 :-)